File: backend/src/merchant/objects/apiTypes/CreateProductRequest.scala
package delivery.merchant.objects.apiTypes

import delivery.merchant.objects.{ProductBundleGroup, ProductInventoryMode}
import delivery.domain.{ListingStatus, MerchantId}

final case class CreateProductRequest(
    merchantId: MerchantId,
    name: String,
    description: String,
    imageUrl: String,
    categoryName: String,
    price: Double,
    remainingStock: Int,
    listingStatus: ListingStatus,
    inventoryMode: Option[ProductInventoryMode] = None,
    maxPerOrder: Option[Int] = None,
    bundleGroups: Option[List[ProductBundleGroup]] = None
)

File: frontend/src/objects/merchant/apiTypes/CreateProductRequest.ts
import type { ListingStatus, MerchantId } from '@/objects/shared/ids'
import type { ProductBundleGroup } from '@/objects/merchant/Product'

export interface CreateProductRequest {
  merchantId: MerchantId
  name: string
  description: string
  imageUrl: string
  categoryName: string
  price: number
  remainingStock: number
  listingStatus: ListingStatus
  bundleGroups?: ProductBundleGroup[]
}
